src/core/Group.cpp

```diff
@@ -1079,6 +1123,82 @@ bool Group::resolveAutoTypeEnabled() const
     }
 }
 
+bool Group::resolveBrowserIntegrationHideEntriesEnabled() const
+{
+    switch (m_data.browserIntegrationHideEntriesEnabled) {
+    case Inherit:
+        if (!m_parent) {
+            return false;
+        } else {
+            return m_parent->resolveBrowserIntegrationHideEntriesEnabled();
+        }
+    case Enable:
+        return true;
+    case Disable:
+        return false;
+    default:
+        Q_ASSERT(false);
+        return false;
+    }
+}
+
+bool Group::resolveBrowserIntegrationSkipAutoSubmitEnabled() const
+{
+    switch (m_data.browserIntegrationSkipAutoSubmitEnabled) {
+    case Inherit:
+        if (!m_parent) {
+            return false;
+        } else {
+            return m_parent->resolveBrowserIntegrationSkipAutoSubmitEnabled();
+        }
+    case Enable:
+        return true;
+    case Disable:
+        return false;
+    default:
+        Q_ASSERT(false);
+        return false;
+    }
+}
+
+bool Group::resolveBrowserIntegrationOnlyHttpAuthEnabled() const
+{
+    switch (m_data.browserIntegrationOnlyHttpAuthEnabled) {
+    case Inherit:
+        if (!m_parent) {
+            return false;
+        } else {
+            return m_parent->resolveBrowserIntegrationOnlyHttpAuthEnabled();
+        }
+    case Enable:
+        return true;
+    case Disable:
+        return false;
+    default:
+        Q_ASSERT(false);
+        return false;
+    }
+}
+
+bool Group::resolveBrowserIntegrationNotHttpAuthEnabled() const
```

Comment:
I'd much prefer a generic function that checks tri-state values in the custom data up the hierarchy.  You just need to pass in the custom data attribute name to make this generic and eliminate 8 functions. You should also use the standard customdata()->contains(...) for the singular group check instead of a custom function just for browser.